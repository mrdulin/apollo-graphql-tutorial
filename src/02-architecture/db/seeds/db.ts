import * as Knex from 'knex';
import faker from 'faker';

export async function seed(knex: Knex): Promise<any> {
  await knex('addresses').insert([
    {
      address_id: 1,
      address_city: faker.address.city(),
      address_country: faker.address.country(),
      address_state: faker.address.state(),
    },
    {
      address_id: 2,
      address_city: faker.address.city(),
      address_country: faker.address.country(),
      address_state: faker.address.state(),
    },
  ]);
  await knex('users').insert([
    { user_id: 1, user_nme: faker.name.findName(), user_email: faker.internet.email(), user_address_id: 1 },
    { user_id: 2, user_nme: faker.name.findName(), user_email: faker.internet.email(), user_address_id: 1 },
    { user_id: 3, user_nme: faker.name.findName(), user_email: faker.internet.email(), user_address_id: 2 },
  ]);

  const posts: any[] = [];
  for (let i = 0; i < 10; i++) {
    const post = {
      post_title: faker.lorem.sentence(),
      post_content: faker.lorem.paragraphs(5),
      post_author_id: 1,
    };
    posts.push(post);
  }
  await knex('posts').insert(posts);
}
