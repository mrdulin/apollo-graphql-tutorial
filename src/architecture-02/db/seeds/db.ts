import * as Knex from 'knex';
import faker from 'faker';

export async function seed(knex: Knex): Promise<any> {
  await knex('addresses').insert([
    {
      address_id: 1,
      address_city: faker.address.city(),
      address_country: faker.address.country(),
      address_state: faker.address.state(),
    },
    {
      address_id: 2,
      address_city: faker.address.city(),
      address_country: faker.address.country(),
      address_state: faker.address.state(),
    },
  ]);
  await knex('users').insert([
    { user_id: 1, user_nme: faker.name.findName(), user_email: faker.internet.email(), user_address_id: 1 },
    { user_id: 2, user_nme: faker.name.findName(), user_email: faker.internet.email(), user_address_id: 1 },
    { user_id: 3, user_nme: faker.name.findName(), user_email: faker.internet.email(), user_address_id: 2 },
  ]);
}
