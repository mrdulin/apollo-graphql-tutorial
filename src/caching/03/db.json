{
  "users": [
    {
      "id": 1,
      "name": "Dr. Amya Schiller",
      "email": "Eda4@yahoo.com"
    },
    {
      "id": 2,
      "name": "Consuelo Mohr",
      "email": "Adele_Ankunding99@yahoo.com"
    },
    {
      "id": 3,
      "name": "Pete Gorczany",
      "email": "Geovanny_Satterfield@gmail.com"
    }
  ],
  "store": {
    "1": "aaa",
    "2": {
      "name": "Raegan Armstrong"
    },
    "3": []
  }
}