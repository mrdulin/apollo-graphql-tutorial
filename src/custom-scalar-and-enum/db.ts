enum Device {
  UNKNOWN = 'Other',
  DESKTOP = 'Computers',
  HIGH_END_MOBILE = 'Mobile devices with full browsers',
  TABLET = 'Tablets with full browsers',
  CONNECTED_TV = 'Devices streaming video content to TV screens',
}

export const db = {
  campaignPerformanceReports: [
    {
      campaignId: 1,
      campaignNme: 'test',
      device: Device.DESKTOP,
    },
  ],
};
