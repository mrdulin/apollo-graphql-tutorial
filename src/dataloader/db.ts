import faker from 'faker';

interface IUser {
  id: number;
  name: string;
  email: string;
}

interface IPost {
  id: number;
  title: string;
  content: string;
  userId: number;
}

const users: IUser[] = [
  { id: 1, name: 'a', email: 'a@gmail.com' },
  { id: 2, name: 'b', email: 'b@gmail.com' },
  { id: 3, name: 'c', email: 'c@gmail.com' },
];

const posts: IPost[] = [
  { id: 1, title: faker.lorem.sentence(), content: faker.lorem.paragraph(), userId: 1 },
  { id: 2, title: faker.lorem.sentence(), content: faker.lorem.paragraph(), userId: 1 },
  { id: 3, title: faker.lorem.sentence(), content: faker.lorem.paragraph(), userId: 1 },
];

const db = { users, posts };
export { db, IUser, IPost };
