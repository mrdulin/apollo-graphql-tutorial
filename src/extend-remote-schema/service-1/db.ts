const db = {
  posts: [],
};

export { db };
