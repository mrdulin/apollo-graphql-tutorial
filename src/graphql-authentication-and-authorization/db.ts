import faker from 'faker';

enum Role {
  admin = 'admin',
  viewer = 'viewer',
  editor = 'editor',
}

const db = {
  users: [
    {
      id: 1,
      name: faker.name.findName(),
      email: faker.internet.email(),
      role: Role.admin,
      bitcoinAddress: faker.finance.bitcoinAddress(),
    },
    {
      id: 2,
      name: faker.name.findName(),
      email: faker.internet.email(),
      role: Role.viewer,
      bitcoinAddress: faker.finance.bitcoinAddress(),
    },
  ],

  posts: [
    { id: 1, title: faker.lorem.sentence(), content: faker.lorem.paragraph(), authorId: 1 },
    { id: 2, title: faker.lorem.sentence(), content: faker.lorem.paragraph(), authorId: 1 },
    { id: 3, title: faker.lorem.sentence(), content: faker.lorem.paragraph(), authorId: 2 },
  ],
};

export { db, Role };
