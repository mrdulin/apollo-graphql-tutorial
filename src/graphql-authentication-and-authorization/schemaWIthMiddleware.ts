import { applyMiddleware } from 'graphql-middleware';
import { makeExecutableSchema } from 'graphql-tools';

import { authMiddleware } from './middleware';
import { resolvers } from './resolvers';
import { typeDefs } from './typeDefs';

const schema = makeExecutableSchema({
  typeDefs,
  resolvers,
});

const schemaWithMiddleware = applyMiddleware(schema, authMiddleware);
export { schemaWithMiddleware as schema };
