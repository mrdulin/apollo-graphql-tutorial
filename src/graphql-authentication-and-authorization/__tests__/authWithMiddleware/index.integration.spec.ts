import { createTestClient } from 'apollo-server-testing';
import { schema } from '../../schemaWIthMiddleware';
import { ApolloServer } from 'apollo-server';
import * as Q from '../queries';
import * as M from '../mutations';
import { context0, context1, context2, context3 } from '../contexts';
import { Role } from '../../db';

describe('auth with middleware', () => {
  describe('Query.user', () => {
    it.each`
      context     | gql                         | operationName                   | name
      ${context0} | ${Q.user}                   | ${'userById'}                   | ${'no user found, required auth: "viewer:editor:admin"'}
      ${context1} | ${Q.user}                   | ${'userById'}                   | ${'user.role: "viewer", required auth: "viewer:editor:admin"'}
      ${context2} | ${Q.user}                   | ${'userById'}                   | ${'user.role: "editor", required auth: "viewer:editor:admin"'}
      ${context3} | ${Q.user}                   | ${'userById'}                   | ${'user.role: "admin", required auth: "viewer:editor:admin"'}
      ${context1} | ${Q.userWithBitcoinAddress} | ${'userByIdWithBitcoinAddress'} | ${'user.role: "viewer", required auth: "viewer:editor:admin", bitcoinAddress require auth: "admin"'}
    `(`$name`, async ({ context, gql, operationName }) => {
      const server = new ApolloServer({ schema, context });
      const { query } = createTestClient(server);
      const actualValue = await query({ query: gql, operationName, variables: { id: 1 } });
      expect(actualValue).toMatchSnapshot();
    });
  });

  describe(`Query.adminUsers | required roles: ${Role.admin}`, () => {
    const adminUsersOperationName = 'adminUsers';
    it.each`
      context     | gql             | operationName              | name
      ${context0} | ${Q.adminUsers} | ${adminUsersOperationName} | ${'no user found'}
      ${context1} | ${Q.adminUsers} | ${adminUsersOperationName} | ${'user.role: "viewer"'}
      ${context2} | ${Q.adminUsers} | ${adminUsersOperationName} | ${'user.role: "editor"'}
      ${context3} | ${Q.adminUsers} | ${adminUsersOperationName} | ${'user.role: "admin"'}
    `(`$name`, async ({ context, gql, operationName }) => {
      const server = new ApolloServer({ schema, context });
      const { query } = createTestClient(server);
      const actualValue = await query({ query: gql, operationName });
      expect(actualValue).toMatchSnapshot();
    });
  });

  describe(`Mutation.createPost | required roles: ${Role.admin}`, () => {
    const createPostOperationName = 'createPost';
    it.each`
      context     | gql             | operationName              | name
      ${context0} | ${M.createPost} | ${createPostOperationName} | ${'no user found'}
      ${context1} | ${M.createPost} | ${createPostOperationName} | ${'user.role: "viewer"'}
      ${context2} | ${M.createPost} | ${createPostOperationName} | ${'user.role: "editor"'}
      ${context3} | ${M.createPost} | ${createPostOperationName} | ${'user.role: "admin"'}
    `(`$name`, async ({ context, gql, operationName }) => {
      const server = new ApolloServer({ schema, context });
      const { query } = createTestClient(server);
      const createPostInput = {
        title: 'jest',
        content: 'jest content',
      };
      const actualValue = await query({ query: gql, operationName, variables: { input: createPostInput } });
      expect(actualValue).toMatchSnapshot();
    });
  });
});
