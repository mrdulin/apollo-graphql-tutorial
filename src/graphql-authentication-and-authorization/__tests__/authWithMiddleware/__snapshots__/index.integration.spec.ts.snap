// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth with middleware Mutation.createPost | required roles: admin no user found 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: no permission],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`auth with middleware Mutation.createPost | required roles: admin user.role: "admin" 1`] = `
Object {
  "data": Object {
    "createPost": Object {
      "code": 0,
      "message": "ok",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`auth with middleware Mutation.createPost | required roles: admin user.role: "editor" 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: no permission],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`auth with middleware Mutation.createPost | required roles: admin user.role: "viewer" 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: no permission],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`auth with middleware Query.adminUsers | required roles: admin no user found 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: no permission],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`auth with middleware Query.adminUsers | required roles: admin user.role: "admin" 1`] = `
Object {
  "data": Object {
    "adminUsers": Array [
      Object {
        "email": "elsa@example.com",
        "id": "2",
        "name": "elsa",
        "role": "admin",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`auth with middleware Query.adminUsers | required roles: admin user.role: "editor" 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: no permission],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`auth with middleware Query.adminUsers | required roles: admin user.role: "viewer" 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: no permission],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`auth with middleware Query.user no user found, required auth: "viewer:editor:admin" 1`] = `
Object {
  "data": Object {
    "user": null,
  },
  "errors": Array [
    [GraphQLError: no permission],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`auth with middleware Query.user user.role: "admin", required auth: "viewer:editor:admin" 1`] = `
Object {
  "data": Object {
    "user": Object {
      "email": "mrdulin@example.com",
      "id": "1",
      "name": "mrdulin",
      "role": "viewer",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`auth with middleware Query.user user.role: "editor", required auth: "viewer:editor:admin" 1`] = `
Object {
  "data": Object {
    "user": Object {
      "email": "mrdulin@example.com",
      "id": "1",
      "name": "mrdulin",
      "role": "viewer",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`auth with middleware Query.user user.role: "viewer", required auth: "viewer:editor:admin" 1`] = `
Object {
  "data": Object {
    "user": Object {
      "email": "mrdulin@example.com",
      "id": "1",
      "name": "mrdulin",
      "role": "viewer",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`auth with middleware Query.user user.role: "viewer", required auth: "viewer:editor:admin", bitcoinAddress require auth: "admin" 1`] = `
Object {
  "data": Object {
    "user": Object {
      "bitcoinAddress": null,
      "email": "mrdulin@example.com",
      "id": "1",
      "name": "mrdulin",
      "role": "viewer",
    },
  },
  "errors": Array [
    [GraphQLError: no permission],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
